internal/middlewares/bearer_auth.go

```diff
@@ -44,10 +45,21 @@ func BearerAuthenticateFunc(key interface{}, _ *zap.Logger) func(string) (interf
 					}
 				}
 			}
+			isEmailConfirmed := false
```

Comment:
To improve the readability and maintainability of the code, it may be better to break the current function into smaller functions, each with a single responsibility. For example, you could move the logic for creating the "auth" and "role" variables into separate functions.
Alternatively, consider using typed claims instead of a map. This approach would eliminate the need for type assertions and make the code more readable and maintainable.

Reply:
Done